staging/src/k8s.io/apiserver/pkg/util/version/version.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"fmt"
+	"strings"
+	"sync/atomic"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/version"
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+	baseversion "k8s.io/component-base/version"
+)
+
+var Effective MutableEffectiveVersions = newEffectiveVersion()
```

Comment:
It looks like every caller of this could use an instance passed through options and the normal configuration process. Which current or future consumer needed this to be a mutable singleton?